test/tests/jobs/CancelJobTest.cpp

```diff
@@ -206,28 +227,86 @@ struct CancelJobTest : tpunit::TestFixture {
         ASSERT_EQUAL(result[0][0], "PAUSED");
     }
 
-    // Cancel a job
-    void cancelJob() {
-        // Create a job
+    // Cancel a child job
+    void cancelChildJob() {
+        // Create a parent job
         SData command("CreateJob");
-        command["name"] = "job";
+        command["name"] = "parent";
         STable response = tester->executeWaitVerifyContentTable(command);
-        string jobID = response["jobID"];
+        string parentID = response["jobID"];
+
+        // Get the parent
+        command.clear();
+        command.methodLine = "GetJob";
+        command["name"] = "parent";
+        tester->executeWaitVerifyContent(command);
+
+        // Create the child
+        command.clear();
+        command.methodLine = "CreateJob";
+        command["name"] = "child";
+        command["parentJobID"] = parentID;
+        response = tester->executeWaitVerifyContentTable(command);
+        string childID = response["jobID"];
 
-        // Cancel it
+        // Create a sibling
+        command.clear();
+        command.methodLine = "CreateJob";
+        command["name"] = "sibling";
+        command["parentJobID"] = parentID;
+        response = tester->executeWaitVerifyContentTable(command);
+        string siblingID = response["jobID"];
+
+        // Finish the parent to put the child in the QUEUED state
+        command.clear();
+        command.methodLine = "FinishJob";
+        command["jobID"] = parentID;
+        tester->executeWaitVerifyContent(command);
+
+        // Try cancelling one of the children, but none of them are running so it should fail.
         command.clear();
         command.methodLine = "CancelJob";
-        command["jobID"] = jobID;
+        command["jobID"] = childID;
+        tester->executeWaitVerifyContent(command, "404 Invalid jobID - Cannot cancel a job that has no RUNNING siblings");
```

Comment:
You don't need to test this here, you test this in `cancelJobWithoutSiblings`

Reply:
Copy/paste lol